core/stablecog.py

```diff
@@ -15,7 +17,45 @@
 from core import viewhandler
 from core import settings
 from core import settingscog
+from threading import Thread
 
+async def update_progress(event_loop, status_message_task, s, queue_object, tries):
+    status_message = status_message_task.result()
+    try:
+        progress_data = s.get(url=f'{settings.global_var.url}/sdapi/v1/progress').json()
+
+        if progress_data["current_image"] is None and tries <= 10:
+            time.sleep(3)
+            event_loop.create_task(update_progress(event_loop, status_message_task, s, queue_object, tries + 1))
+            return
+
+        if progress_data["current_image"] is None and tries > 10:
+            return
+
+        image = Image.open(io.BytesIO(base64.b64decode(progress_data["current_image"])))
+
+        with contextlib.ExitStack() as stack:
+            buffer = stack.enter_context(io.BytesIO())
+            image.save(buffer, 'PNG')
+            buffer.seek(0)
+            file = discord.File(fp=buffer, filename=f'{queue_object.seed}.png')
+
+        ips = round((int(queue_object.steps) - progress_data["state"]["sampling_step"]) / progress_data["eta_relative"], 2)
```

Comment:
`progress_data["eta_relative"]` can potentially be `0` here which causes a ZeroDivisionError. I'm seeing this occur on my bot -- this might be a difference between stable-diffusion-webui and SD.Next (which I use) but there's no reason not to check for this.

Recommend replacing with:

```python
        ips = '?'
        if progress_data["state"]["sampling_step"] != 0:
            ips = round((int(queue_object.steps) - progress_data["state"]["sampling_step"]) / progress_data["eta_relative"], 2)
```

Reply:
I appreciate your hint! Maybe you can create a PR?



> SD.Next (which I use)

I used it too ([here's my PR for this bot to work on SD.Next as Vladmandic recommended](https://github.com/Kilvoctu/aiyabot/commit/d47d3ab9bfa7c3edf19ac0a6dfdc035ac5f89111)), but many my lovely extensions doesn't work with it